### PR TITLE
deep frying nonfoods is no longer a qdel v2

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
this pr was denied through a qwerty moment about a year ago, even though most devs were in agreeance on it
https://github.com/BeeStation/BeeStation-Hornet/pull/2624
lets try it again

## Why It's Good For The Game
deepfrying is a wayyyy to easily done way to qdel an item, and doesn't even check resistance flags

## Changelog
:cl:
del: removes deepfrying items to destroy them. you can now eat them out of the crust, or just clean them off
/:cl:

### CODE REVIEW

**Comment by Archanial** [comment]
This makes deep fryer not do anything.

Unless you make stuff get damaged or associate frying level level with damage/complete consumption of said item this is not going to fly.

File: code/modules/food_and_drinks/food/snacks_bread.dm [hunk 1]
```diff
@@ -245,20 +245,21 @@
...
 		fried.reagents.trans_to(src, fried.reagents.total_volume)
 		qdel(fried)
 	else
 		fried.forceMove(src)
 
-/obj/item/reagent_containers/food/snacks/deepfryholder/Destroy()
-	if(contents)
-		QDEL_LIST(contents)
-	. = ..()
+/obj/item/reagent_containers/food/snacks/deepfryholder/proc/clean_batter()
```
Comment (Archanial) [inline code/modules/food_and_drinks/food/snacks_bread.dm:256]: Missing signal handler

**Comment by KubeRoot** [comment]
> this pr was denied through a qwerty moment

image

More like maintainers decided to hold a vote and the playerbase voted against it.

**Comment by ike709** [comment]
\> pretending votes matter

**Comment by AnCopper** [comment]
> > this pr was denied through a qwerty moment
> 
> image
> 
> More like maintainers decided to hold a vote and the playerbase voted against it.

Golden existed back then.

**Comment by williamcll** [comment]
> > > this pr was denied through a qwerty moment
> > 
> > 
> > image
> > More like maintainers decided to hold a vote and the playerbase voted against it.
> 
> Golden existed back then.

pretty sure more people voted than the amount of people that played golden.

**Comment by ike709** [comment]
\> arguing about the irrelevant votes

**Comment by zeskorion** [comment]
> This makes deep fryer not do anything.
> 
> Unless you make stuff get damaged or associate frying level level with damage/complete consumption of said item this is not going to fly.

This makes the deep fryer not do anything other than make deep fried food, yes.

**Review by francinum — CHANGES_REQUESTED** [review]
Add the signal handler mark and ping me for a merge.

Ideally this'd tie into the damage system, but the ability to make any item ireecoverable regardless of resistance flags with one click is kinda a major issue.

**Comment by Archanial** [comment]
🔑

**Comment by Archanial** [comment]
Okay I don't entirely oppose the idea but this pr is flawed.

Items should get damaged, damage to the item scaling on the time thing is fried or on fry level and possibly completely destroyed if it's hp is low enough and is fried enough.

And then, based on amount of hp gone, you should add nutrients to deepfry holder so we don't end up with deepfry, take bite, deepfry take bite...

**Comment by zeskorion** [comment]
> Okay I don't entirely oppose the idea but this pr is flawed.
> 
> Items should get damaged, damage to the item scaling on the time thing is fried or on fry level and possibly completely destroyed if it's hp is low enough and is fried enough.
> 
> And then, based on amount of hp gone, you should add nutrients to deepfry holder so we don't end up with deepfry, take bite, deepfry take bite...

i don't believe the deepfryer should be able to destroy items in any capacity. they're way too easy to access
and, there's no reason to have it deal damage to an item it doesnt destroy. damaging items is a very, very rare mechanic. its easier to just haul it to the recycler at that point, yknow?

**Review by francinum — APPROVED** [review]
Review requirements fulfilled.

**Comment by PowerfulBacon** [comment]
The resistance flags are checked when something is deep fried.
Also can't you infinitely eat stuff?

**Comment by zeskorion** [comment]
> The resistance flags are checked when something is deep fried. Also can't you infinitely eat stuff?

it checks the indestructable tag when you try to put it in the deep fryer. the deep fryer doesnt check any other flags, like heatproof
and spraying cooking oil doesnt check any flags

**Comment by AnCopper** [comment]
@PowerfulBacon

**Comment by MCterra10** [comment]
Based?

**Comment by AnCopper** [comment]
Should've been done a year ago, finally.

**Comment by zeskorion** [comment]
> Should've been done a year ago, finally.

First made on october the fourth, last year

**Pull request merged**